Introduce guidelines for closing stale issues/PRs (#7332)
pytest-dev/pytest: CONTRIBUTING.rst

   * Delete the PR body, it usually contains a duplicate commit message.

Handling stale issues/PRs
-------------------------

Stale issues/PRs are those where pytest contributors have asked for questions/changes

> and the authors didn't got around to answer/implement them yet after a somewhat long time, or

Reviewer: ```suggestion
and the authors didn't get around to answer/implement them yet after a somewhat long time, or
```

> the discussion simply died because people seemed to lost interest.

Reviewer: ```suggestion
the discussion simply died because people seemed to lose interest.
```

There are many reasons why people don't answer questions or implement requested changes:

> they might get busy, lost interest, or just forgot about it,

Reviewer: ```suggestion
they might get busy, lose interest, or just forget about it,
```

but the fact is that this is very common in open source software.

The pytest team really appreciates every issue and pull request, but being a high-volume project
with many issues and pull requests being submitted daily, we try to reduce the number of stale
issues and PRs by regularly closing them. When an issue/pull request is closed in this manner,
it is by no means a dismissal of the topic being tackled by the issue/pull request, but it

> is just a way for us to clear up the queue and make the maintainers work more manageable. Submitters

Reviewer: ```suggestion
is just a way for us to clear up the queue and make the maintainers' work more manageable. Submitters
```

can always reopen the issue/pull request in their own time later if it makes sense.

When to close
~~~~~~~~~~~~~

Here are a few general rules the maintainers use to decide when to close issues/PRs because
of lack of inactivity:

> * Issues labeled ``question``: closed after 14 days inactive.

Reviewer 2: ```suggestion
* Issues labeled ``question`` or ``needs information``: closed after 14 days inactive.
```

Many, though not all, of [these issues](https://github.com/pytest-dev/pytest/issues?q=is%3Aissue+is%3Aopen+label%3A%22status%3A+needs+information%22+sort%3Aupdated-asc) could be closed as stale - in most, we're waiting on the reporter to tell us what's going on.

* Issues labeled ``proposal``: closed after six months inactive.
* Pull requests: after one month, consider pinging the author, update linked issue, or consider closing. For pull requests which are nearly finished, the team should consider finishing it up and merging it.

The above are **not hard rules**, but merely **guidelines**, and can be (and often are!) reviewed on a case-by-case basis.

Closing pull requests
~~~~~~~~~~~~~~~~~~~~~

When closing a Pull Request, it needs to be acknowledge the time, effort, and interest demonstrated by the person which submitted it. As mentioned previously, it is not the intent of the team to dismiss stalled pull request entirely but to merely to clear up our queue, so a message like the one below is warranted when closing a pull request that went stale:

    Hi <contributor>,

    First of all we would like to thank you for your time and effort on working on this, the pytest team deeply appreciates it.

>     We noticed it has been awhile since you have updated this PR, however. pytest is a high activity project, with may issues/PRs being opened daily, so it is hard for us maintainers to track which PRs are ready for merging, for review, or need more attention.

Reviewer: ```suggestion
    We noticed it has been awhile since you have updated this PR, however. pytest is a high activity project, with many issues/PRs being opened daily, so it is hard for us maintainers to track which PRs are ready for merging, for review, or need more attention.
```

>     So for those reasons we think it is best to close the PR for now, but with the only intention to cleanup our queue, it is by no means a rejection of your changes. We still encourage you to re-open this PR (it is just a click of a button away) when you are ready to get back to it.

Reviewer: might want to double check whether someone can reopen their own PR that a maintainer closes -- they might not be able to (and may need to comment) -- that would just change some of the prose here

Author: Indeed I'm not sure. If somebody knows for sure, I will be happy to change it here. 👍
